core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnUniqueness.java

```diff
@@ -364,6 +381,25 @@ public Boolean areColumnsUnique(Intersect rel, RelMetadataQuery mq,
         return false;
       }
 
+      if (Aggregate.isSimple(rel)) {
```

Comment:
5 levels of nesting might be a bit much, I know this is kind of a vague comment but could you check if there's any opportunities for early returns or just a way of splitting this up?